### ISSUE TITLE
Option for `lines-before-block` to ignore single-line jsdoc comments

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [gajus/eslint-plugin-jsdoc#1295]
## Motivation

When documenting parameters, simple single-line descriptions can be enough. For example:
```ts
/** The parameters for a request */
export type RequestParams = {
	/** The year to retrieve. */
	year: `${number}`;
	/** The month to retrieve. */
	month: `${number}`;
}
```
In this case, I don't think newlines before the jsdoc comments add any value.

## Current behavior

The rule warns even with `ignoreSameLine: true`, as the jsdoc is on a separate line.

## Desired behavior

Don't warn if the jsdoc fits on a single line.

## Alternatives considered

* Don't warn if the jsdoc fits on a single line *and* only contains a description. That would be enough for the example above, but another use case is providing type information for a variable on the next line in javascript, and I don't think requiring a preceding newline helps much there either.

**Comment by JoshuaKGoldberg** [comment]
IMO this option would be best enabled by default. This succinct / one-line style is pretty common in projects, and it's surprising for many to see the linter report against it.

**Comment by thernstig** [comment]
Although, if it was a `/** @type {string[]} */` e.g. for a top-level variable I would expect it to have a line break before.

**Comment by ehoogeveen-medweb** [comment]
I'm not sure I would want to enforce it:
```js
function foo() {
  /** @type { string[] | undefined } */
  let myStringArray;
  /** @type { number[] | undefined } */
  let myNumberArray;
  // ...
}
```
(assuming also https://github.com/gajus/eslint-plugin-jsdoc/issues/1296) looks reasonable to me.

**Comment by JoshuaKGoldberg** [comment]
Checking in, sorry for the ping @gajus. I just want to confirm - would you be open to a PR that makes checking single-line JSDoc comments an opt-in option?

**Comment by brettz9** [comment]
@JoshuaKGoldberg : SGTM!

**Comment by brettz9** [comment]
I've submitted #1408 with the option on by default. (Let me know if you want that changed.)

@JoshuaKGoldberg : Have time to review?
